### PR TITLE
Adding monikerized return type

### PR DESCRIPTION
**Pull request by joelmartinez**
Re-opening this from this previous PR from a fork:
https://github.com/docascode/ECMA2Yaml/pull/110

@TianqiZhang, I've changed the data structure so that description doesn't have to be duplicated amongst all individual values. Additionally, return type now has its own data model across the process ... and it is backwards compatible with the "legacy" single return value pipelines.

### CODE REVIEW

File: ECMA2Yaml/ECMAHelper/ECMALoader.cs
Comment (TianqiZhang) [inline ECMA2Yaml/ECMAHelper/ECMALoader.cs]: we can directly use `ECMALoader.LoadFrameworkAlternate()` to parse framework alternate values. #Resolved

File: ECMA2Yaml/ECMAHelper/YamlConverters/SDPYamlConverter.Delegate.cs
Comment (TianqiZhang) [inline ECMA2Yaml/ECMAHelper/YamlConverters/SDPYamlConverter.Delegate.cs:24]: can these logic be shared between delegate and member converters? #Resolved

**Comment by TianqiZhang** [comment]
@joelmartinez the integration test files are located at test folder. We have three sets of tests, MREF, SDP and UWP_SDP. You can run the script `/CI/devopsIntTest.ps1` to run the integration test locally and compare the files to find the diff :) #Resolved

File: ECMA2Yaml/ECMAHelper/Models/VersionedValue.cs
Comment (TianqiZhang) [inline ECMA2Yaml/ECMAHelper/Models/VersionedValue.cs:33]: is `VersionedTypeReference` used?

File: ECMA2Yaml/ECMAHelper/ECMALoader.cs
Comment (joelmartinez) [reply ECMA2Yaml/ECMAHelper/ECMALoader.cs]: Ahh good tip, I've simplified the code to use this #Resolved

File: ECMA2Yaml/ECMAHelper/YamlConverters/SDPYamlConverter.Delegate.cs
Comment (joelmartinez) [reply ECMA2Yaml/ECMAHelper/YamlConverters/SDPYamlConverter.Delegate.cs:24]: I'd considered that even in my first iteration ... but this section of the code effectively has _two_ return types, depending on if the value is monikerized or not (ie. setting .Returns, or ReturnsWithMoniker). So to avoid having a helper method with complicated out/ref semantics or something else, I just effectively copied these two routines in the two places where it was used. #Resolved

File: ECMA2Yaml/ECMAHelper/Models/VersionedValue.cs
Comment (joelmartinez) [reply ECMA2Yaml/ECMAHelper/Models/VersionedValue.cs:33]: Yes, but this question made me realize that I wasn't fully propagating the value to the end yaml ... it is now :) So this adds "refType" to the yaml, which is not something that we're rendering today, but we will have to at some point since we have `ref returns` in C#. I didn't want to mess with the single return output, so I didn't change that data type to include it (it's using TypeReference right now); but this at least gives us an easier line of evolution going forward once we decide to implement ref returns.

**Comment by joelmartinez** [comment]
@TianqiZhang yep, found those integration test scripts, and ran them before uploading these latest edits :)  #Resolved

**Comment by TianqiZhang** [comment]
@joelmartinez I think this PR is ready to merge, however we'd better wait for the docs-ui change to release to prod before we merge ECMA2Yaml change, so that the extra properties in yaml doesn't cause schema validation errors.

**Review by TianqiZhang — APPROVED** [review]

**Pull request merged**
